lib/Dialects/LinalgTransform/Transforms/ScopedTransform.cpp

```diff
@@ -0,0 +1,84 @@
+//===-- ScopedTransform.cpp -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Dialects/LinalgTransform/ScopedTransform.h"
+#include "mlir/Transforms/InliningUtils.h"
+
+using namespace mlir;
+
+namespace {
+struct Rewriter : public PatternRewriter {
+  Rewriter(MLIRContext *ctx) : PatternRewriter(ctx) {}
+};
+} // namespace
+
+linalg::transform::ScopeOp linalg::transform::wrapInScope(Operation *op) {
+  Rewriter rewriter(op->getContext());
+  rewriter.setInsertionPoint(op);
+
+  auto scope = rewriter.create<linalg::transform::ScopeOp>(
+      op->getLoc(), op->getResultTypes(), op->getOperands());
+  Region &body = scope.body();
+  rewriter.setInsertionPointToStart(&body.emplaceBlock());
+  BlockAndValueMapping bv;
+  bv.map(op->getOperands(), body.addArguments(op->getOperandTypes()));
+
+  Operation *cloneInScope = op->clone(bv);
+  rewriter.insert(cloneInScope);
+  rewriter.setInsertionPointToEnd(&body.front());
```

Comment:
Strictly speaking, this is unnecessary. The insertion point will be pointing there anyway.